Sources/PointFree/Admin/FreeEpisodes.swift

```diff
@@ -12,35 +12,12 @@ import Prelude
 import Styleguide
 import Tuple
 import View
+import Views
 
-let indexFreeEpisodeEmailMiddleware =
+let indexFreeEpisodeEmailMiddleware: Middleware<StatusLineOpen, ResponseEnded, Tuple1<User?>, Data> =
   requireAdmin
     <| writeStatus(.ok)
-    >=> respond(freeEpisodeView.contramap(lower))
-
-private let freeEpisodeView = View<User> { _ in
-  [
-    h2(["Send free episode email"]),
-
-    ul(
-      Current.episodes()
-        .filter((!) <<< ^\.subscriberOnly)
-        .sorted(by: their(^\.sequence))
-        .map(li <<< freeEpisodeEmailRowView.view)
-    )
-  ]
-}
-
-private let freeEpisodeEmailRowView = View<Episode> { ep in
-  [
-    p([
-      .text(ep.title),
-      form([action(path(to: .admin(.freeEpisodeEmail(.send(ep.id))))), method(.post)], [
-        input([type(.submit), value("Send email!")])
-        ])
-      ])
-  ]
-}
+    >=> respond(freeEpisodeView(episodes: Current.episodes(), today: Current.date()))
```

Comment:
I experimented with converting this view to a function instead of `View`. it works, we just need a new `respond` middleware to deal with it, which I haven't done yet. this view didnt need any custom data from the route, so I could do it more simply by just calling the function immediately.